internal/productcatalog/postgresadapter/feature.go

```diff
@@ -0,0 +1,164 @@
+package postgresadapter
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/openmeterio/openmeter/internal/productcatalog"
+	"github.com/openmeterio/openmeter/internal/productcatalog/postgresadapter/ent/db"
+	db_feature "github.com/openmeterio/openmeter/internal/productcatalog/postgresadapter/ent/db/feature"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+// Adapter implements remote connector interface as driven port.
+type featureDBAdapter struct {
+	logger *slog.Logger
+	db     *db.Client
+}
+
+func NewPostgresFeatureDBAdapter(db *db.Client, logger *slog.Logger) productcatalog.FeatureDBConnector {
+	return &featureDBAdapter{
+		db:     db,
+		logger: logger,
+	}
+}
+
+func (c *featureDBAdapter) CreateFeature(ctx context.Context, feature productcatalog.DBCreateFeatureInputs) (productcatalog.Feature, error) {
+	query := c.db.Feature.Create().
+		SetName(feature.Name).
+		SetKey(feature.Key).
+		SetNamespace(feature.Namespace).
+		SetMeterSlug(feature.MeterSlug)
+
+	if feature.MeterGroupByFilters != nil {
+		query = query.SetMeterGroupByFilters(*feature.MeterGroupByFilters)
+	}
+
+	entity, err := query.
+		Save(ctx)
+
+	if err != nil {
+		return productcatalog.Feature{}, err
+	}
+
+	return mapFeatureEntity(entity), nil
+}
+
+func (c *featureDBAdapter) FindByKey(ctx context.Context, namespace string, key string, includeArchived bool) (*productcatalog.Feature, error) {
```

Comment:
Get by id and get by key can be merged. It usually helps to handle featureIdOrKey usecases. Like:

```
func (r Repository) GetByIdOrSlug(ctx context.Context, orgId string, idOrSlug string) (*db.Subject, error) {
	return r.ent.Subject.Query().
		WithMeter().
		Where(
                         Where(db_feature.Namespace(namespace)),
			entreport.Or(
				entreport.ID(idOrSlug),
				entreport.Slug(idOrSlug),
			),
		).
		Only(ctx)
}
```

Reply:
i think in general having two unique identifiers being used is confusing and a bad design. if we use keys then we could / should manage all user initiated logic with keys IMO (that would make IDs obsolete for the most part), or at least be more intentional about when to use which and why.